### PR TITLE
DHSCFT-116: Make api call the database

### PR DESCRIPTION
**Pull request by adawari-thomas-bjss**
# Description

Jira ticket: [DHSCFT-116](https://bjss-enterprise.atlassian.net/browse/DHSCFT-116)

Include a summary of what this pull request is changing and any relevant background context.

## Changes

- Moved service and repository logic to HealthData module
- Wrote tests to validate repository, service, and controller class behaviours

## Validation

API is returning data from the database
image

### CODE REVIEW

File: api/DHSC.FingertipsNext.Modules.HealthData.UnitTests/Repository/HealthDataRepositoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                Guid.NewGuid().ToString()
+            );
+
+        _dbContext = new HealthDataDbContext(dbOptions.Options);
+    }
+
+    [Fact]
+    public void RepositoryInitialization_ShouldThrowError_IfNullDBContextIsProvided()
+    {
+        var act = () => _repository = new HealthDataRepository(null);
```
Comment (williamwillis11) [inline api/DHSC.FingertipsNext.Modules.HealthData.UnitTests/Repository/HealthDataRepositoryTests.cs:27]: my ide and github is throwing the 'Cannot convert null literal to non-nullable reference type.' warning - can this be resolved?

File: api/DHSC.FingertipsNext.Modules.HealthData.UnitTests/Services/IndicatorServiceTests.cs [hunk 2]
```diff
@@ -1,65 +1,145 @@
...
                 1,
-                ArgEx.IsEquivalentTo<string[]>(["a10", "a11", "a12", "a13", "a14", "a15", "a16", "a17", "a18", "a19"]),
-                ArgEx.IsEquivalentTo<int[]>(Enumerable.Range(20, 10).ToArray())
+                ArgEx.IsEquivalentTo<string[]>(expectedAreaCodes),
+                ArgEx.IsEquivalentTo<int[]>(expectedYears)
             );
     }
 
     [Fact]
-    public async Task GetIndicatorData_PassesDistinctFiltersToProvider_WhenSuppliedDuplicates()
```
Comment (williamwillis11) [inline api/DHSC.FingertipsNext.Modules.HealthData.UnitTests/Services/IndicatorServiceTests.cs:48]: do we need a duplicates test?

Comment (adawari-thomas-bjss) [reply api/DHSC.FingertipsNext.Modules.HealthData.UnitTests/Services/IndicatorServiceTests.cs:48]: The `GetIndicatorData_PassesFirst10DistinctFiltersToRepository_WhenSuppliedMore` test that's been added also tests against duplicates
